command/secrets/output/params.go

```diff
@@ -0,0 +1,248 @@
+package output
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/0xPolygon/polygon-edge/command"
+	"github.com/0xPolygon/polygon-edge/secrets"
+	"github.com/0xPolygon/polygon-edge/secrets/helper"
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+const (
+	dataDirFlag   = "data-dir"
+	configFlag    = "config"
+	validatorFlag = "validator"
+	blsFlag       = "bls"
+	nodeIDFlag    = "node-id"
+)
+
+var (
+	params = &outputParams{}
+)
+
+var (
+	errInvalidConfig   = errors.New("invalid secrets configuration")
+	errInvalidParams   = errors.New("no config file or data directory passed in")
+	errUnsupportedType = errors.New("unsupported secrets manager")
+)
+
+type outputParams struct {
+	dataDir    string
+	configPath string
+
+	outputNodeID    bool
+	outputValidator bool
+	outputBLS       bool
+
+	secretsManager secrets.SecretsManager
+	secretsConfig  *secrets.SecretsManagerConfig
+
+	validatorAddress string
+	blsPubkey        string
+
+	nodeID string
+}
+
+func (op *outputParams) validateFlags() error {
+	if op.dataDir == "" && op.configPath == "" {
+		return errInvalidParams
+	}
+
+	return nil
+}
+
+func (op *outputParams) outputSecrets() error {
+	if err := op.initSecretsManager(); err != nil {
+		return err
+	}
+
+	if err := op.initValidatorAddress(); err != nil {
+		return err
+	}
+
+	if err := op.initBLSPublicKey(); err != nil {
+		return err
+	}
+
+	return op.initNodeID()
+}
+
+func (op *outputParams) initSecretsManager() error {
+	if op.hasConfigPath() {
+		return op.initFromConfig()
+	}
+
+	return op.initLocalSecretsManager()
+}
+
+func (op *outputParams) hasConfigPath() bool {
+	return op.configPath != ""
+}
+
+func (op *outputParams) initFromConfig() error {
+	if err := op.parseConfig(); err != nil {
+		return err
+	}
+
+	var secretsManager secrets.SecretsManager
+
+	switch op.secretsConfig.Type {
+	case secrets.HashicorpVault:
+		vault, err := helper.SetupHashicorpVault(op.secretsConfig)
+		if err != nil {
+			return err
+		}
+
+		secretsManager = vault
+	case secrets.AWSSSM:
+		AWSSSM, err := helper.SetupAWSSSM(op.secretsConfig)
+		if err != nil {
+			return err
+		}
+
+		secretsManager = AWSSSM
+	case secrets.GCPSSM:
+		GCPSSM, err := helper.SetupGCPSSM(op.secretsConfig)
+		if err != nil {
+			return err
+		}
+
+		secretsManager = GCPSSM
+	default:
+		return errUnsupportedType
+	}
```

Comment:
Could you create a helper function to return secret manager by config because both `command/secrets/init` and `command/secrets/output` package use

Reply:
Yup! Good idea.

Reply:
I gave it a shot with https://github.com/0xPolygon/polygon-edge/pull/705/commits/98cf7ffd20f578b36bd3127d3c095a83c09a4fca 

I think more can be done for a larger effort later down the road.